test/github-issues/6168/issue-6168.ts

```diff
@@ -0,0 +1,115 @@
+import "reflect-metadata";
+import {Connection} from "../../../src/connection/Connection";
+import {closeTestingConnections, createTestingConnections, reloadTestingDatabases} from "../../utils/test-utils";
+import {Table} from "../../../src/schema-builder/table/Table";
+import { QueryRunner, createConnection } from "../../../src";
+import { MysqlConnectionOptions } from "../../../src/driver/mysql/MysqlConnectionOptions";
+import { expect } from "chai";
+
+const questionName = "question";
+const categoryName = "category";
+
+const createDB = async (queryRunner: QueryRunner, dbName: string) => {
+    await queryRunner.createDatabase(dbName, true);
+    const questionTableName = `${dbName}.${questionName}`;
+    const categoryTableName = `${dbName}.${categoryName}`;
+
+    await queryRunner.createTable(new Table({
+        name: questionTableName,
+        columns: [
+            {
+                name: "id",
+                type: "int",
+                isPrimary: true,
+                isGenerated: true,
+                generationStrategy: "increment"
+            },
+            {
+                name: "name",
+                type: "varchar",
+            }
+        ],
+    }), true);
+
+    await queryRunner.createTable(new Table({
+        name: categoryTableName,
+        columns: [
+            {
+                name: "id",
+                type: "int",
+                isPrimary: true,
+                isGenerated: true,
+                generationStrategy: "increment"
+            },
+            {
+                name: "questionId",
+                type: "int",
+            }
+        ],
+        foreignKeys: [
+            {
+                columnNames: ["questionId"],
+                referencedTableName: questionTableName,
+                referencedColumnNames: ["id"],
+                name: "FK_CATEGORY_QUESTION"
+            }
+        ]
+    }), true);
+};
+
+describe("github issues > #6168 fix multiple foreign keys with the same name in a mysql multi-tenanted DB", () => {
+
+    let connections: Connection[];
+    let testConnections = [] as Connection[];
+    let testQueryRunners = [] as QueryRunner[];
+    before(async () => {
+        connections = await createTestingConnections({
+            entities: [__dirname + "/entity/*{.js,.ts}"],
+            enabledDrivers: ["mysql"],
+            schemaCreate: false,
+            dropSchema: false,
+        });
+
+        if (connections.length > 0) {
+            const queryRunner = connections[0].createQueryRunner();
```

Comment:
Since you already have to have an extra line here, could this instead be -

```suggestion
        for (const connection of connections) {
          const queryRunner = connection.createQueryRunner();
```

Same for the other similar ones like this.

Right now it's just the 1 connection but if you do this it makes splitting them up easier (eg, adding maria to this)